OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -60,6 +61,9 @@ public class ArmamentInfo : PausableConditionalTraitInfo, Requires<AttackBaseInf
 		[Desc("Palette to render Muzzle flash sequence in")]
 		public readonly string MuzzlePalette = "effect";
 
+		[Desc("Custom offset used as base for calculating muzzle overlay position that is independent from LocalOffset.")]
+		public readonly WVec[] MuzzleOverlayOffset = Array.Empty<WVec>();
```

Comment:
```suggestion
		public readonly WVec[] MuzzleOverlayOffsets = Array.Empty<WVec>();
```
Is this offset-per-barrel? If so the description here should specify that because it wasn't clear to me.

Reply:
I used singular because `LocalOffset` also uses singular. If the new field is going to use plural (`MuzzleOverlayOffsets`) then `LocalOffset` should also become `LocalOffsets` (or not)?

Reply:
I noticed both points about `LocalOffset` but decided not to scope creep. If you want to add another commit that changes (IMO fixes) that then that'd be great. If not - ok, but I don't think the old one being wrong should be a reason for the new one to also be wrong.

Reply:
Well that would be scope creep, because it would require creating `UpdateRule`, right?

Reply:
Yes, which is why I only suggested changing the one that is being added now.
Whether to change the preexisting one or not (by adding an update rule) is up to you.